public/app/plugins/datasource/stackdriver/types.ts

```diff
@@ -21,39 +21,70 @@ export enum MetricFindQueryTypes {
   Aggregations = 'aggregations',
   Aligners = 'aligners',
   AlignmentPeriods = 'alignmentPeriods',
+  Selectors = 'selectors',
+  SLOServices = 'sloServices',
+  SLO = 'slo',
 }
 
 export interface VariableQueryData {
   selectedQueryType: string;
   metricDescriptors: MetricDescriptor[];
   selectedService: string;
   selectedMetricType: string;
+  selectedSLOService: string;
   labels: string[];
   labelKey: string;
   metricTypes: Array<{ value: string; name: string }>;
   services: Array<{ value: string; name: string }>;
   projects: Array<{ value: string; name: string }>;
+  sloServices: Array<{ value: string; name: string }>;
   projectName: string;
 }
 
-export interface StackdriverQuery extends DataQuery {
+export enum QueryType {
+  METRICS = 'metrics',
+  SLO = 'slo',
+}
+
+export const queryTypes = [
+  { label: 'Metrics', value: QueryType.METRICS },
+  { label: 'Service Level Objectives (SLO)', value: QueryType.SLO },
+];
+
+export interface MetricQuery {
   projectName: string;
   unit?: string;
   metricType: string;
-  service?: string;
-  refId: string;
   crossSeriesReducer: string;
   alignmentPeriod?: string;
-  perSeriesAligner: string;
+  perSeriesAligner?: string;
   groupBys?: string[];
   filters?: string[];
   aliasBy?: string;
-  metricKind: string;
-  valueType: string;
-  datasourceId?: number;
+  metricKind?: string;
+  valueType?: string;
   view?: string;
 }
 
+export interface SLOQuery {
+  projectName: string;
+  alignmentPeriod?: string;
+  perSeriesAligner?: string;
+  aliasBy?: string;
+  selectorName: string;
+  serviceId: string;
+  sloId: string;
+  goal?: number;
```

Comment:
What is `goal` for? I don't see a respective field in [SLOQuery](https://github.com/grafana/grafana/blob/b7d7d5b5fa993585fcb3fb85158cbfcd74cda8ff/pkg/tsdb/stackdriver/types.go#L37) in the backend.

Reply:
Correct! 
When defining an SLO you have to specify a goal, i.e a level that you don't want to go under. See the goal being displayed as a dotted line in Stackdriver. I wanted to save the goal in the query model so that we can implement this easily when the graph panel is converted to react and it has support for displaying dotted lines like this. The new graph panel will probably be released in a few months of time.
<img width="303" alt="Skärmavbild 2020-03-24 kl  10 01 40" src="https://user-images.githubusercontent.com/2388950/77407488-7e614200-6db6-11ea-8e5f-429eef0c2648.png">

Reply:
Maybe a comment about that will make this more obvious for anyone else looking at the code?